server/main.cpp

```diff
@@ -217,29 +171,50 @@ int main(int argc, char *argv[])
         }
     }
 
-    QStringList filterRules;
+
+    /* The logging rules will be evaluated sequentially
+     *  1. All debug categories off
+     *  2. The stored categories from the nymead.conf will be appended
+     *  3. The Command line parameters will be added (-p and -d)
+     *  4. QT_LOGGING_CONF
+     *  5. QT_LOGGING_RULES
+     *
+     * The final filter rules will be set.
+     */
+
+    QStringList loggingRules;
+    loggingRules << "*.debug=false";
+    // Load the rules from nymead.conf file and append them to the rules
+    NymeaSettings nymeaSettings(NymeaSettings::SettingsRoleGlobal);
+    nymeaSettings.beginGroup("LoggingRules");
+    foreach (const QString &category, nymeaSettings.childKeys()) {
+        loggingRules << QString("%1=%2").arg(category).arg(nymeaSettings.value(category, "false").toString());
+    }
+    nymeaSettings.endGroup();
+
+    // Append the rules depending on the application parameter
     if (parser.isSet(allOption)) {
-        filterRules << "*.debug=true";
-        filterRules << "*Traffic.debug=false";
-        filterRules << "*Debug.debug=false";
-    } else {
-        filterRules << "*.debug=false";
+        loggingRules << "*.debug=true";
+        loggingRules << "*Traffic.debug=false";
+        loggingRules << "*Debug.debug=false";
```

Comment:
I think you're changing the logic a bit here...

now, if someone sets TcpServerTraffic.debug=true in nymead.conf and then also adds the -p option, it will turn off the *Traffic.debug again.

It should probably be:

```
    QStringList loggingRules;
    if (parser.isSet(allOption)) {	
        filterRules << "*.debug=true";
        filterRules << "*Traffic.debug=false";
        filterRules << "*Debug.debug=false";
    } else {
        loggingRules << "*.debug=false";
    }
    // Load the rules from nymead.conf file and append them to the rules
    NymeaSettings nymeaSettings(NymeaSettings::SettingsRoleGlobal);
    nymeaSettings.beginGroup("LoggingRules");
    foreach (const QString &category, nymeaSettings.childKeys()) {
        loggingRules << QString("%1=%2").arg(category).arg(nymeaSettings.value(category, "false").toString());
    }
    nymeaSettings.endGroup();
```

This way, the order would be:

-p
nymead.conf
-d

which I think makes sense to deprioritize -p a bit...